### PR TITLE
Add linting config and fix issues.

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title. Prep work to roll out golangci-lint actions across the board.

### CODE REVIEW

File: metrics/prometheus_exporter_test.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func expectPromSrv(t *testing.T, expectedAddr string) {
 	}
 }
 
+// nolint // TODO: Test is currently skipped.
```
Comment (mattmoor) [inline metrics/prometheus_exporter_test.go:71]: Add the issue number to unskip (for grep)?

File: testutils/clustermanager/perf-tests/pkg/benchmark.go [hunk 2]
```diff
@@ -40,7 +40,7 @@ const (
 )
 
 // backupLocations are used in retrying cluster creation, if stockout happens in one location.
-// TODO(chizhg): it's currently not used, use it in the cluster creation retry logic.
+// nolint // TODO(chizhg): it's currently not used, use it in the cluster creation retry logic.
```
Comment (mattmoor) [inline testutils/clustermanager/perf-tests/pkg/benchmark.go:43]: @chizhg is there an issue for this?

File: metrics/prometheus_exporter_test.go (hunk 1 above)
Comment (markusthoemmes) [reply metrics/prometheus_exporter_test.go]: Inlined it instead, even prettier.

**Pull request merged**